### PR TITLE
Added QGC complex mission item importing

### PR DESCRIPTION
**Pull request by Jonsen94**
Currently the import_qgroundcontrol_mission function can only import simple items (waypoints).

QGC also supports surveys, corridor scans and structure scans.

This pull should fix the first two and allow importing surveys and corridor scans.

See https://github.com/mavlink/MAVSDK/issues/1116

It's not tested as I have problems connecting the server with my python-mavsdk scripts. (Timeout issues).
But it compiles at least.

### CODE REVIEW

**Comment by Jonsen94** [comment]
Okay what am I missing.
I don't see where the added code fails in the test ?

File: src/plugins/mission/mission_impl.cpp [hunk 1]
```diff
@@ -1024,6 +1024,49 @@ Mission::Result MissionImpl::build_mission_items(
...
+    const Json::Value& json_complex_mission_item,
+    MissionItem& new_mission_item)
+{
+    if (json_complex_mission_item["TransectStyleComplexItem"].isNull()) {
+        LogWarn() << "Unknown complex item type (" << json_complex_mission_item["complexItemType"]
+                  << ")";
+        return Mission::Result::UnsupportedMissionCmd;
+    }
+
+    Json::Value complex_item = json_complex_mission_item["TransectStyleComplexItem"];
```
Comment (julianoes) [inline src/plugins/mission/mission_impl.cpp:1059]: Could you add a code comment for the reader on what this complex "TransectStyleComplexItem" actually is?

**Review by julianoes — CHANGES_REQUESTED** [review]
Thanks, I think the change is nice and clean.

However, it would be great if you could add a unit test for such a complex mission file.
See https://github.com/mavlink/MAVSDK/blob/develop/src/plugins/mission/mission_import_qgc_test.cpp

**Comment by Jonsen94** [comment]
Btw for documentation reasons if someone finds this pr:
Taken from the GQC git (https://github.com/mavlink/qgroundcontrol):
Complex Mission Item (which is a super class of Surveys, Corridor Scans and Structure Scans):  [ComplexMissionItem.h](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/ComplexMissionItem.h)
Transect Mission Item (which is a subclass of the Complex Mission Item and a super class of Surveys and Corridor Scans, *note*: not of the structure scans): [TransectStyleComplexItem.h](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/TransectStyleComplexItem.h)
The json key "Items" for the array I used for getting the simple mission items: [TransectStyleComplexItem.cc#L38](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/TransectStyleComplexItem.cc#L38)

Other complex items:
[StructureScanComplexItem.h](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/StructureScanComplexItem.h)
[CorridorScanComplexItem.h](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/CorridorScanComplexItem.h)
[SurveyComplexItem.h](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/SurveyComplexItem.h)

File: src/plugins/mission/mission_impl.cpp (hunk 1 above)
Comment (Jonsen94) [reply src/plugins/mission/mission_impl.cpp:1064]: I added a comment and a test case.
The tests output lots of warnings (unsupported mission items) for 206 (MAV_CMD_DO_SET_CAM_TRIGG_DIST) and 20 (MAV_CMD_NAV_RETURN_TO_LAUNCH ) which are used in the test plan file.

Comment (julianoes) [reply src/plugins/mission/mission_impl.cpp:1064]: Ok thanks. I guess that's expected because they are not understood by the simple MAVSDK mission API.

**Review by julianoes — APPROVED** [review]
Thanks for the PR and follow up.

**Pull request merged**